src/application/logic/Explosion.java

```diff
@@ -0,0 +1,58 @@
+package application.logic;
+
+import application.GameController;
+import application.sharedObject.IRenderable;
+import application.sharedObject.RenderableHolder;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+
+import java.util.ArrayList;
+
+public class Explosion implements IRenderable {
+    private double x ,y ;
+    private int changedSprite = 0 ;
+    private long previousChange ;
+    protected boolean destroyed = false ;
```

Comment:
**IRenderable interface already has destroyed field**
- use this.destroyed instead
- set the initial value of fields in the constructor